helix-term/src/commands/lsp.rs

```diff
@@ -1367,13 +1368,25 @@ fn compute_inlay_hints_for_view(
                         None => continue,
                     };
 
-                let label = match hint.label {
-                    lsp::InlayHintLabel::String(s) => s,
-                    lsp::InlayHintLabel::LabelParts(parts) => parts
-                        .into_iter()
-                        .map(|p| p.value)
-                        .collect::<Vec<_>>()
-                        .join(""),
+                let label = {
+                    let mut label = match hint.label {
+                        lsp::InlayHintLabel::String(s) => s,
+                        lsp::InlayHintLabel::LabelParts(parts) => parts
+                            .into_iter()
+                            .map(|p| p.value)
+                            .collect::<Vec<_>>()
+                            .join(""),
+                    };
+                    // Truncate the hint if too long
+                    if let Some(limit) = inlay_hints_length_limit {
+                        let limit = limit.into();
+                        if label.len() > limit {
```

Comment:
The byte length of the string works ok for ASCII but this should be checking width from `UnicodeWidthStr` to figure out how many Unicode grapheme clusters can fit within the limit.